src/scriptdev2/scripts/eastern_kingdoms/shadowfang_keep/shadowfang_keep.cpp

```diff
@@ -598,8 +598,11 @@ struct boss_arugalAI : public ScriptedAI
         {
             if (!m_bAttacking)
             {
-                DoCastSpellIfCan(m_creature->getVictim(), SPELL_VOID_BOLT);
-                m_uiVoidboltTimer = urand(2900, 4800);
+                if (Unit* victim = m_creature->getVictim())
```

Comment:
StopAttacking() is a custom function in this script.
It shouldn't impact the spell casting.

Actually that function should be called StopMeleeAttack and StartMeleeAttack, because it allows the creature to start or stop movement to the target.

Anyway, that script looks pretty old. I think it was one of the firsts to be developed, so the logic of the event looks kind of weird to me.

However, I still think it shouldn't crash on spell cast.
Usually the target pointer during the CastSpell function is checked multiple times, and this wasn't a problem before.